SAK-31641 Switch to VARCHAR rather than INTs for enum.

buckett opened the pull request:
This makes the oauth code use VARCHARs rather than INTs for it’s enums as it makes the database more readable and helps to prevent problems when enums are changed.

This is also needed as we currently have this setup from our 10.x OAuth deployment and migrating to INTs seems like a backwards step.

buckett commented:
If someone knows the Oracle code for this that would be helpful.

steveswinsburg commented:
I'll add the oracle SQL as a comment.

steveswinsburg commented:
Here you go @buckett:

````
ALTER TABLE oauth_accessors 
MODIFY (
  status VARCHAR2(255)
, type VARCHAR2(255)
);

UPDATE oauth_accessors 
SET status = CASE
  WHEN status = 0 THEN 'VALID'
  WHEN status = 1 THEN 'REVOKED'
  WHEN status = 2 THEN 'EXPIRED'
END;

UPDATE oauth_accessors 
SET type = CASE
  WHEN type = 0 THEN 'REQUEST'
  WHEN type = 1 THEN 'REQUEST_AUTHORISING'
  WHEN type = 2 THEN 'REQUEST_AUTHORISED'
  WHEN type = 3 THEN 'ACCESS'
END;
{code}
````

Any reason why the MySQL does the first alter statement to `int`, then to `varchar`? I haven't a MySQL schema to check what it was before.

buckett commented:
Good spot on the the rouge statement, dropped (original alter).

Brilliant, thanks Steve, I've updated the PR, slight reformat so that I could easily see the differences between the MySQL/Oracle.
